main/streams/streams.c

```diff
@@ -1706,7 +1706,7 @@ PHPAPI int php_register_url_stream_wrapper(const char *protocol, php_stream_wrap
 		return FAILURE;
 	}
 
-	return zend_hash_str_add_ptr(&url_stream_wrappers_hash, protocol, protocol_len, wrapper) ? SUCCESS : FAILURE;
+	return zend_hash_add_ptr(&url_stream_wrappers_hash, zend_string_init_interned(protocol, protocol_len, 1), wrapper) ? SUCCESS : FAILURE;
```

Comment:
Seems this part is ok. User space wrappers runs through php_register_url_stream_wrapper_volatile() and save into FG(stream_wrappers). So far i couldn't find a code path where php_register_url_stream_wrapper() is called outside the startup phase. Thus names of those saved into url_stream_wrappers_hash need to be interned, as they can interact with the runtime code.